crates/iceberg-catalog/src/service/task_queue/mod.rs

```diff
@@ -0,0 +1,85 @@
+use async_trait::async_trait;
+use chrono::Utc;
+use sqlx::FromRow;
+use std::time::Duration;
+use uuid::Uuid;
+
+pub mod delete_queue;
+pub mod tabular_expiration_queue;
+
+#[async_trait]
+pub trait TaskQueue: std::fmt::Debug {
+    type Task: Send + Sync + 'static;
+    type Input: Send + Sync + 'static;
+
+    fn queue_name(&self) -> &'static str;
+    async fn poll(&self) -> crate::api::Result<Option<Self::Task>>;
```

Comment:
Would pick_task_for_execution maybe be a better name?
We might want to add a poll_status or so in the future to wait for a task to add its terminal state?
Poll sounds to me read_only - which is why I suggested to have a separate read_pool somewhere else. But this polling actually picks a task, increasing the attempts and everything.